server/src/sas/FormatOnTypeProvider.ts

```diff
@@ -164,6 +152,69 @@ export class FormatOnTypeProvider {
     }
   }
 
+  private _getExpectedCurLineIndent(
+    referBlockStartLine: boolean,
+    foldingBlock: FoldingBlock | null,
+    line: number,
+    tabSize: number,
+  ) {
+    const lastNotEmptyLine = this._getLastNotEmptyLine(line - 1);
+    if (
+      referBlockStartLine &&
+      (foldingBlock || lastNotEmptyLine !== undefined)
+    ) {
+      const blockStartLine = foldingBlock?.startLine ?? lastNotEmptyLine!;
+      const blockStartLineText = this.model.getLine(blockStartLine);
+      return this._getIndentLength(blockStartLineText, tabSize);
+    }
+
+    const tokens = this.syntaxProvider.getSyntax(line);
+    const cleanedTokens = this._cleanTokens(line, tokens);
+    const index = this._findSemicolonTokenRightToLeft(
+      line,
+      cleanedTokens,
+      cleanedTokens.length - 1,
+    );
+    if (index > 0) {
```

Comment:
```suggestion
    if (index <= 0) {
       return;
    }
```
Can we update this if to be written the opposite way, and unindent the remaining code?